### ISSUE TITLE
Legacy syntax shows cryptic error message

### ISSUE THREAD

**Issue by Stephanevg** [pester/Pester#1376]
Hi,

This is more for tracking purposes.  

I do have the following test.

```powershell
Describe 'My Describe block' {

    it  'should be Negative'  {
        $true | should be $False
    }

    it 'should be positive' {
        $true | should be $true
    }
}

```

And tried changing to the following one afterwards

```powershell

Describe -Name 'My Describe block' -Fixture {

    it -name 'should be Negative' -test {
        $true | should be $False
    }

    it -name 'should be positive' -test {
        $true | should be $true
    }
}

```

but in both cases, I get the following results:

```powershell
>  invoke-pester C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\simpletest.tests.ps1
Starting test discovery in 1 files.
Discovering tests in C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\simpletest.tests.ps1.
Found 2 tests. 308ms
Test discovery finished. 335ms
Running tests from 'C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\simpletest.tests.ps1'

Describing My Describe block
    [-] should be Negative 56ms (41ms|15ms)
      ParameterBindingException: Parameter set cannot be resolved using the specified named parameters.
      at <ScriptBlock>, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\simpletest.tests.ps1:4
      at <ScriptBlock>, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:1392
      at <ScriptBlock>, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:1351
      at Invoke-ScriptBlock, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:1512
      at Invoke-TestItem, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:589
      at Invoke-Block, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:262
      at <ScriptBlock>, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:333
      at <ScriptBlock>, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:1392
      at <ScriptBlock>, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:1351
      at Invoke-ScriptBlock, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:1515
      at Invoke-Block, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:362
      at Run-Test, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:1129
      at Invoke-Test, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:1834
      at Invoke-Pester<End>, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\Pester.psm1:729
    [-] should be positive 102ms (62ms|40ms)
      ParameterBindingException: Parameter set cannot be resolved using the specified named parameters.
      at <ScriptBlock>, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\simpletest.tests.ps1:8
      at <ScriptBlock>, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:1392
      at <ScriptBlock>, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:1351
      at Invoke-ScriptBlock, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:1512
      at Invoke-TestItem, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:589
      at Invoke-Block, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:262
      at <ScriptBlock>, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:333
      at <ScriptBlock>, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:1392
      at <ScriptBlock>, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:1351
      at Invoke-ScriptBlock, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:1515
      at Invoke-Block, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:362
      at Run-Test, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:1129
      at Invoke-Test, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\new-runtimepoc\Pester.Runtime.psm1:1834
      at Invoke-Pester<End>, C:\Users\taavast3\OneDrive\Repo\Projects\OpenSource\pesterPR\Pester\Pester.psm1:729
Tests completed in 0ms
Tests Passed: 0, Failed: 2, Skipped: 0, Pending: 0, Inconclusive: 0
```

**Comment by Stephanevg** [comment]
I downloaded the alpha4 from the gallery and actually ahve the same error as above

**Comment by nohwnd** [comment]
I guess it is the legacy syntax of Should? But that error is hardly helpful. This needs improvement.

**Comment by nohwnd** [comment]
It is the legacy syntax of Should. I added the code that consumes it back in there and throw error pointing to the migration guide when I detect legacy syntax. 

![image](https://user-images.githubusercontent.com/5735905/71471062-a6003880-27ce-11ea-8c85-a01510b8a0e0.png)

**Comment by nohwnd** [comment]
This is not easy to fix, the parameter sets won't resolve correctly when adding the legacy parameter set only to be able to throw and exeption as shown in #1406. 

@renehernandez  maybe you have some thoughts on this?

**Comment by fflaten** [comment]
Fixed in #1561 ? Error is now similar to image above. Anything else remaining?
